invenio_vocabularies/contrib/names/resources.py

```diff
@@ -52,3 +61,40 @@ def name_resolve_by_id(self):
         )
 
         return item.to_dict(), 200
+
+    @request_extra_args
+    @request_search_args
+    @response_handler(many=True)
+    @login_required
```

Comment:
minor: shouldn't we instead be using the permission policy and the `AuthenticatedUser` need? If this is how this is done in other places, feel free to ignore :)

Reply:
That's a good point! 
It really depends on whether we want to protect just the endpoint, as the login_required decorator does, or if we also want to secure the service methods. I believe we should go with the latter approach, so I’ll make the necessary changes based on your suggestion. Thanks!